src/components/donation-flow/steps/authentication/Authentication.tsx

```diff
@@ -0,0 +1,47 @@
+import { useSession } from 'next-auth/react'
+import { Box, Checkbox, Radio, Typography } from '@mui/material'
+
+import RadioAccordionGroup from '../../common/RadioAccordionGroup'
+import InlineLoginForm from './InlineLoginForm'
+import InlineRegisterForm from './InlineRegisterForm'
+import { useEffect } from 'react'
+import { useFormikContext } from 'formik'
+
+export default function Authentication() {
+  const { data: session } = useSession()
+  const formik = useFormikContext()
+  useEffect(() => {
+    if (session?.user) {
+      formik.setFieldValue('authentication', undefined)
+    }
+  }, [session?.user])
```

Comment:
@dimitur2204 I'm not able to follow the logic here, Shouldn't the field `authentication` be one of the following 

```tsx
export enum DonationFormDataAuthState {
  LOGIN = 'login',
  REGISTER = 'register',
  ANONYMOUS = 'anonymous',
  AUTHENTICATED = 'authenticated',
}
```

I don't get why do we set it to `undefined` if session user is present.

I thought it should be set to `DonationFormDataAuthState.AUTHENTICATED`

Reply:
I agree this shouldn't be like that and I will change it.

I am actually unsure about that flow as a whole. I think the "without register"/"anonymous" checkbox should probably be outside of that as a separate checkbox, available to both logged in and guest users. 

I am waiting on response from the design team if they have any other suggestions on that

Reply:
It doesn't make sense to be authenticated and select the option without registration.

The option "Without registration" actually should be having a different meaning, which is to control if the donation is listed as public or private. 

Let's double confirm that with @igoychev and @Juicebgs

Reply:
I actually see a point in an Anonymous donation even if you are registered, but I am unsure if we want to do that.

Reply:
Just for reference this is how the whole flow looks like with the latest changes with a checkbox I was expermienting with:
![full-flow-after-auth](https://user-images.githubusercontent.com/61479393/213940987-153c0721-267c-4438-9816-ddbd26ba707e.gif)

Reply:
We have the following user story: As a donor, I would like to donate and receive my donation certificate and yet I would like my name to remain hidden in the donations section. 

That is the reason to have this checkbox. 
Should this checkbox be visible if the visitor is not logged in - no, it will be confusing.

Reply:
@Juicebgs Ok then from what I understand we have 2 checkboxes depending on if you are logged in or not:
* [Logged in] I want to be anonymous 
* [Not logged] I want to continue withour registration

Is that right?